pkgs/development/compilers/flutter/repository.nix

```
@@ -0,0 +1,23 @@
+{ lib, stdenv, dart, fetchurl, src, pname, version, depsSha256 }:
+
+stdenv.mkDerivation {
+  inherit src version;
+
+  pname = "${pname}-deps";
+  buildInputs = [ dart ];
+
+  buildPhase = ''
+    export PUB_CACHE="$out"
+    export FLUTTER_ROOT="$(pwd)"
+    export FLUTTER_TOOLS_DIR="$FLUTTER_ROOT/packages/flutter_tools"
+
+    (cd "$FLUTTER_TOOLS_DIR" && ${dart}/bin/pub get)
```

I am not sure if this causes issues in nixpkgs.

```suggestion
    pushd "$FLUTTER_TOOLS_DIR"
    ${dart}/bin/pub get
    popd
```

---

Actually in this case we don't need to go back to the original directory, so I did only the `pushd` part. But in the `flutter.nix`  did both since it express the intention better.